Verso/PolychromaticSite.lean

```diff
@@ -79,6 +108,72 @@ class B green;
 class H blue
 ```
 
+# The Lovász Local Lemma Approach
+
+The key tool for proving Strauss' conjecture is the *Lovász local lemma*, a powerful result in
+probabilistic combinatorics. The lemma shows that if we have a collection of "bad" events, each of
+which has small probability and is mostly independent of the others, then with positive probability
+*none* of the bad events occur.
+
+## The Symmetric Local Lemma
+
+In its symmetric form, if we have events $`A_1, \ldots, A_n` where each event:
+- has probability at most $`p`, and
+- is independent of all but at most $`d` other events,
+
+then if $`e \cdot p \cdot (d + 1) \leq 1`, there exists an outcome avoiding all bad events.
+
+## Application to Polychromatic Colourings
+
+To prove that a set $`S` of size $`m$ admits a $`k`-polychromatic colouring, we:
+
+1. Consider a random colouring where each integer is coloured uniformly at random from $`k` colours.
+2. For each translate $`n + S` and each colour $`c`, define a "bad event" as the event that no
+   element of $`n + S` receives colour $`c`.
+3. The probability of a bad event is $`(1 - 1/k)^m`, which is small when $`m` is large relative
+   to $`k`.
+4. Bad events for translates $`n + S` and $`n' + S` are independent unless the translates overlap,
+   which bounds the dependency.
+
+Applying the local lemma gives the bound $`m(k) \leq 3k^2`. A more refined analysis using the
+Rado selection principle (a topological compactness argument) extends this to infinite colourings
+and yields the asymptotically optimal bound $`m(k) \leq (3 + o(1)) k \log k`.
```

Comment:
This section misunderstands the application of these results. The Local Lemma works for finitely many translations, and the Rado selection lemma extends it to all translations. Then straightforward analysis gives `m(k) <= 3k^2`, and a more careful asymptotic analysis gives `m(k) <= (3+o(1)) k log k`. This latter bound is not asymptotically optimal (the true value is (1+o(1)) k log k) but it is optimal up to constant factors.